Bottom: e5ace9888afeea175276506f7a008f47451fa76f
Top:    06140fe45f707ae74bfc13611cddcc982dce3366
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-26 03:59:14 +0100

Comment out some calls which can trigger an out-of-context redraw


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 9e4741b..0700837 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -367,7 +367,7 @@ ghid_port_key_press_cb (GtkWidget * drawing_area,
   state = (GdkModifierType) (kev->state);
   mk = ghid_modifier_keys_state (&state);
 
-  ghid_show_crosshair (FALSE);
+  //ghid_show_crosshair (FALSE);
 
   handled = TRUE;		/* Start off assuming we handle it */
   switch (ksym)
@@ -473,7 +473,7 @@ ghid_port_button_press_cb (GtkWidget * drawing_area,
   ghid_note_event_location (ev);
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
-  ghid_show_crosshair (FALSE);
+  //ghid_show_crosshair (FALSE);
   HideCrosshair (TRUE);
   drag = have_crosshair_attachments ();
 
@@ -482,7 +482,7 @@ ghid_port_button_press_cb (GtkWidget * drawing_area,
   ghid_invalidate_all ();
   RestoreCrosshair (TRUE);
   ghid_set_status_line_label ();
-  ghid_show_crosshair (TRUE);
+  //ghid_show_crosshair (TRUE);
   if (!gport->panning)
     g_idle_add (ghid_idle_cb, NULL);
   return TRUE;
@@ -873,7 +873,7 @@ ghid_port_window_leave_cb (GtkWidget * widget,
       cursor_in_viewport = FALSE;
     }
 
-  ghid_show_crosshair (FALSE);
+  //ghid_show_crosshair (FALSE);
   out->has_entered = FALSE;
 
   ghid_screen_update ();
